contrib/submit-queue/submit-queue.go

```diff
@@ -0,0 +1,111 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+// A simple binary for merging PR that match a criteria
+// Usage:
+//   submit-queue -token=<github-access-token> [-min-pr-number=<number>] [-dry-run] [-once]
+//
+// Details:
+// Usage of ./submit-queue:
+//  -alsologtostderr=false: log to standard error as well as files
+//  -dry-run=false: If true, don't actually merge anything
+//  -log_backtrace_at=:0: when logging hits line file:N, emit a stack trace
+//  -log_dir="": If non-empty, write log files in this directory
+//  -logtostderr=false: log to standard error instead of files
+//  -min-pr-number=0: The minimum PR to start with [default: 0]
+//  -one-off=false: If true, only merge one PR, don't run forever
+//  -stderrthreshold=0: logs at or above this threshold go to stderr
+//  -token="": The OAuth Token to use for requests.
+//  -v=0: log level for V logs
+//  -vmodule=: comma-separated list of pattern=N settings for file-filtered logging
+
+import (
+	"errors"
+	"flag"
+
+	"github.com/GoogleCloudPlatform/kubernetes/contrib/submit-queue/github"
+	"github.com/GoogleCloudPlatform/kubernetes/contrib/submit-queue/jenkins"
+
+	"github.com/golang/glog"
+	github_api "github.com/google/go-github/github"
+)
+
+var (
+	token       = flag.String("token", "", "The OAuth Token to use for requests.")
+	minPRNumber = flag.Int("min-pr-number", 0, "The minimum PR to start with [default: 0]")
+	dryrun      = flag.Bool("dry-run", false, "If true, don't actually merge anything")
+	oneOff      = flag.Bool("once", false, "If true, only merge one PR, don't run forever")
+	job         = flag.String("jenkins-job", "kubernetes-e2e-gke-ci", "The job in Jenkins to use for stability testing")
+)
+
+const (
+	user    = "GoogleCloudPlatform"
+	project = "kubernetes"
+)
+
+// This is called on a potentially mergeable PR
+func runE2ETests(client *github_api.Client, pr *github_api.PullRequest, issue *github_api.Issue) error {
+	// Test if the build is stable in Jenkins
+	jenkinsClient := &jenkins.JenkinsClient{Host: "http://kubekins.dls.corp.google.com"}
+	stable, err := jenkinsClient.IsBuildStable(*job)
+	glog.V(2).Infof("Checking build stability")
+	if err != nil {
+		return err
+	}
+	if !stable {
+		glog.Errorf("Build isn't stable, skipping!")
+		return errors.New("Unstable build")
+	}
+	glog.V(2).Infof("Build is stable.")
+	// Ask for a fresh build
+	glog.V(4).Infof("Asking PR builder to build %d", *pr.Number)
+	body := "@k8s-bot test this [testing build queue, sorry for the noise]"
```

Comment:
Added a whitelist.  For now, not worrying about the k8s-bot issues, we can always create a new bot.